src/libmongoc/src/mongoc/mongoc-cluster.c

```diff
@@ -1327,6 +1347,39 @@ _mongoc_cluster_auth_node_x509 (mongoc_cluster_t *cluster,
 #endif
 }
 
+/* True if the client user has requested a specific wire protocol version: */
```

Comment:
```suggestion
/* True if the client user has requested a specific server API version: */
```

I suggest moving these functions definitions to the files associated with the type. Namely:

`mongoc_topology_scanner_uses_server_api` to mongoc-topology-scanner.c
`mongoc_topology_uses_server_api` to mongoc-topology.c
`mongoc_client_uses_server_api` to mongoc-client.c

I suggest removing `mongoc_client_session_uses_server_api` since it is not called.